core/src/test/java/io/lionweb/lioncore/java/serialization/SerializedJsonComparisonUtils.java

```diff
@@ -84,36 +84,46 @@ private static void assertEquivalentNodes(
       throw new AssertionError("(" + context + ") Missing keys found: " + missingKeys);
     }
     for (String key : actualMeaningfulKeys) {
-      if (key.equals("parent")) {
-        assertEquals(
-            "(" + context + ") different parent", expected.get("parent"), actual.get("parent"));
-      } else if (key.equals("concept")) {
-        assertEquals(
-            "(" + context + ") different concept", expected.get("concept"), actual.get("concept"));
-      } else if (key.equals("id")) {
-        assertEquals("(" + context + ") different id", expected.get("id"), actual.get("id"));
-      } else if (key.equals("references")) {
-        assertEquivalentUnorderedArrays(
-            expected.getAsJsonArray("references"),
-            actual.getAsJsonArray("references"),
-            "References of " + context);
-      } else if (key.equals("children")) {
-        assertEquivalentUnorderedArrays(
-            expected.getAsJsonArray("children"),
-            actual.getAsJsonArray("children"),
-            "Children of " + context);
-      } else if (key.equals("properties")) {
-        assertEquivalentUnorderedArrays(
-            expected.getAsJsonArray("properties"),
-            actual.getAsJsonArray("properties"),
-            "Properties of " + context);
-      } else if (key.equals("annotations")) {
-        assertEquivalentUnorderedArrays(
-            expected.getAsJsonArray("annotations"),
-            actual.getAsJsonArray("annotations"),
-            "Annotations of " + context);
-      } else {
-        throw new AssertionError("(" + context + ") unexpected top-level key found: " + key);
+      switch (key) {
+        case "parent":
```

Comment:
Is this supported by Java 8? I did not expect that

Reply:
By this I mean, switch on string constants

Reply:
Yes, all checks passed for this PR.
Even Java 7 supported it: https://docs.oracle.com/javase/8/docs/technotes/guides/language/strings-switch.html